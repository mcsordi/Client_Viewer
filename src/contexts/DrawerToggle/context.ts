import { createContext } from 'react';
import { TNavDrawer } from '../../shared/types/NavDrawer';


type TToggle = {
    show: Boolean
    toggleState: () => void
    drawerOptions: TNavDrawer[]
    handleDrawerOptions: (navDrawer: TNavDrawer[]) => void
}

export const toggle = createContext({} as TToggle)
