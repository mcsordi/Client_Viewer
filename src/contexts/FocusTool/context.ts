import { createContext, Dispatch, SetStateAction } from "react";
type TFocusTool = {
    focus: boolean
    setFocus: Dispatch<SetStateAction<boolean>>
}

export const focusTool = createContext({} as TFocusTool)
