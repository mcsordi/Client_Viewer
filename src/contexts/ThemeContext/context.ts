import { createContext } from "react";

interface IThemeProvider {
    light: boolean,
    changeTheme: () => void
}

export const themeContext = createContext({} as IThemeProvider)

