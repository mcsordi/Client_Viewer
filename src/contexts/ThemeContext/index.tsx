import React, { useCallback, useState } from 'react';
import { themeContext } from './context.ts';
import { Children } from '../../shared/types/Children.ts';

export const ThemeContext: React.FC<Children> = ({ children }) => {
  const [light, setLight] = useState<boolean>(true);
  const changeTheme = useCallback(() => {
    setLight((old) => !old);
  }, []);
  return (
    <themeContext.Provider value={{ light, changeTheme }}>
      {children}
    </themeContext.Provider>
  );
};
