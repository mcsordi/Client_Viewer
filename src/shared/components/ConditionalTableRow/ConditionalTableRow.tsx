import React, { useContext } from 'react';
import { Children } from '../../types/Children';
import { themeContext } from '../../../contexts/ThemeContext/context';

export const ConditionalTableRow: React.FC<Children> = ({ children }) => {
  const { light } = useContext(themeContext);
  return (
    <tr
      className={`${!light && `border-zinc-800`} absolute border border-t-0 border-l-0 border-r-0 last:border-b-0 mr-0 sm:mr-6`}
    >
      <td
        className={`${light ? `bg-white` : `bg-neutral-700`} p-3 relative  w-screen h-12 rounded-md`}
      >
        {children}
      </td>
    </tr>
  );
};
