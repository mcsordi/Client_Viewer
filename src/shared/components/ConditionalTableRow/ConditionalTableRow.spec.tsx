import { render, screen } from '@testing-library/react';
import { ConditionalTableRow } from './ConditionalTableRow';
import { themeContext } from '../../../contexts/ThemeContext/context';

describe('<ConditionalTableRow/>', () => {
  it('should render a component', () => {
    render(<ConditionalTableRow>chicken</ConditionalTableRow>);
    const textTd = screen.getByText('chicken');
    expect(textTd.parentElement).toBeInTheDocument();
  });
  it('should render a component in the dark mode', () => {
    const changeTheme = () => {};
    render(
      <themeContext.Provider value={{ light: false, changeTheme }}>
        <ConditionalTableRow>chicken</ConditionalTableRow>
      </themeContext.Provider>,
    );
    const textTd = screen.getByText('chicken');
    expect(textTd.parentElement).toHaveClass('border-zinc-800');
    expect(textTd.parentElement.firstChild).toHaveClass('bg-neutral-700');
  });
  it('should render a component in the light mode', () => {
    const changeTheme = () => {};
    const { container } = render(
      <themeContext.Provider value={{ light: true, changeTheme }}>
        <ConditionalTableRow>chicken</ConditionalTableRow>
      </themeContext.Provider>,
    );
    expect(container.firstChild).not.toHaveClass('border-zinc-800');
    expect(container.firstChild.firstChild).toHaveClass('bg-white');
  });
  it('should match a snapshot', () => {
    const changeTheme = () => {};
    render(
      <themeContext.Provider value={{ light: false, changeTheme }}>
        <ConditionalTableRow>chicken</ConditionalTableRow>
      </themeContext.Provider>,
    );
    const textTd = screen.getByText('chicken');
    expect(textTd.parentElement).toMatchSnapshot();
  });
});
