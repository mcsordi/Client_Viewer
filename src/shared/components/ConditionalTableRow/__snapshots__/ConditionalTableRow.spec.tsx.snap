// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConditionalTableRow/> should match a snapshot 1`] = `
<tr
  class="border-zinc-800 absolute border border-t-0 border-l-0 border-r-0 last:border-b-0 mr-0 sm:mr-6"
>
  <td
    class="bg-neutral-700 p-3 relative  w-screen h-12 rounded-md"
  >
    chicken
  </td>
</tr>
`;
